oauthproxy.go

```diff
@@ -645,6 +638,13 @@ func (p *OAuthProxy) SignInPage(rw http.ResponseWriter, req *http.Request, code
 		redirectURL = "/"
 	}
 
+	if err := p.ClearSessionCookie(rw, req); err != nil {
+		logger.Printf("Error clearing session cookie: %v", err)
+		p.pageWriter.WriteSignInPage(rw, req, redirectURL, code)
+		return
+	}
+
+	rw.WriteHeader(code)
```

Comment:
From where in the flow is the `SignInPage(...)` method called when you change the session storage (or manipulate the cookie to be invalid).

What would be the statusCode in that case.

Reply:
The status code in this case is 403.

https://github.com/oauth2-proxy/oauth2-proxy/blob/fc701bfd6a992bb074671b91ac89b50a79d3dc3d/oauthproxy.go#L1024